src/libs/Navigation/AppNavigator/ModalStackNavigators.js

```diff
@@ -225,7 +227,26 @@ const SettingsModalStackNavigator = createModalStackNavigator([
         name: 'ReimbursementAccount',
         initialParams: {stepToOpen: CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT},
     },
-]);
+], {
+    state: ({data}) => {
```

Comment:
Yeah, I agree that this code doesn't feel right because it's adding business logic in the wrong layer. Rory, did you try using a higher order component to share this functionality with the workspace pages?

Reply:
The HOC is a good suggestion, giving that a try!

Reply:
> As for solutions, I was thinking we would create a component that would be used by all workspace pages, enclosing the components we have now and would load the full policy then pass it down?

@iwiznia I was struggling to find a way to accomplish this with our current routing configuration, but the HOC might do it!

Reply:
Updated to use a HOC, let me know what you think! I think it's much better now, so thanks for the suggestion!